src/ChangeStream/ResourceStack.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the puli/repository package.
+ *
+ * (c) Bernhard Schussek <bschussek@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Puli\Repository\ChangeStream;
+
+use Puli\Repository\Api\Resource\PuliResource;
+use RuntimeException;
+use Webmozart\Assert\Assert;
+
+/**
+ * Represents a versioned stack of resources for a given path.
+ *
+ * You can access different versions of a resource using a ChangeStream
+ * that will return you resource stacks.
+ *
+ * @since  1.0
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ */
+class ResourceStack
+{
+    /**
+     * @var array
+     */
+    private $stack;
+
+    /**
+     * @param array $stack
+     */
+    public function __construct($stack)
+    {
+        Assert::isArray($stack, 'Built resource stack must be an array.');
+
+        $this->stack = $stack;
+    }
+
+    /**
+     * Get the current version resource.
+     *
+     * @return PuliResource
+     */
+    public function getCurrent()
+    {
+        return $this->get($this->getCurrentVersion());
+    }
+
+    /**
+     * Get the current version number.
+     *
+     * @return int
+     */
+    public function getCurrentVersion()
+    {
+        $versions = $this->getVersions();
```

Comment:
That's something interesting: for you, the first version is 1 and the last is count($stack) ? For the moment I started at 0. What do you think is the most intuitive for developers?